src/distdl/nn/conv_feature.py

```diff
@@ -70,13 +110,44 @@ def __init__(self, P_x, buffer_manager=None, *args, **kwargs):
 
         # Do this before checking serial so that the layer works properly
         # in the serial case
-        self.conv_layer = self.TorchConvType(*args, **kwargs)
+        self.conv_layer = self.TorchConvType(in_channels=in_channels,
+                                             out_channels=out_channels,
+                                             kernel_size=kernel_size,
+                                             stride=stride,
+                                             padding=0,
+                                             padding_mode='zeros',
+                                             dilation=dilation,
+                                             groups=groups,
+                                             bias=bias)
 
         self.serial = False
         if self.P_x.size == 1:
             self.serial = True
             return
 
+        dims = len(self.P_x.shape)
+
+        # We will be using global padding to compute local padding,
+        # so expand it to a numpy array
+        padding = np.atleast_1d(padding)
+        padding = np.pad(padding,
+                         pad_width=(dims-len(padding), 0),
+                         mode='constant',
+                         constant_values=0)
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.kernel_size = kernel_size
+        self.stride = stride
+        self.padding = padding
```

Comment:
This padding variable includes the batch and channel dimensions, now.   Perhaps it should be set before lines 132-136, to preserve the torch interface?

Reply:
I think the Torch interface preserves whatever was input to __init__, right? We still need the expanded padding in setup, so we can call it "self.global_padding" instead.

Reply:
This works for me, I see you did this already